### DISCUSSION TITLE
Best practices for updating GitHub Apps

### DISCUSSION THREAD

**Discussion by wassson** [community/community#69037] [Other Feature Feedback, Questions, & Ideas]
I am developing a GitHub App and I'm wondering if there are best practices related to updating a GH App's functionality in repos that already have the app installed. 

Let's say I develop a new feature that requires a new, custom `label`. A solution that comes to mind is to have the users reinstall the GH App on their repo and when the `installation_repositories.added` event triggers I can create that label. For obvious reasons this is cumbersome and is the wrong approach, but I'm not finding any good answers to this question in my research. I guess I could also write a script that grabs all installations and makes a request to create the label for each repo, but that feels invasive.

How are people handling this kind of change to their apps?

**Comment by queenofcorgis** [comment]
Welcome to the GitHub Community, @austinwasson , we're happy you're here! You are more likely to get a useful response if you are posting your question in the applicable category. I've gone ahead and moved it for you. Good luck!

**Reply by wassson** [reply]
Thank you!
